### PR TITLE
[feature-wip][array-type] Array data can be loaded in stream load.

### LINKED ISSUE #8367: [feature-wip][array-type] Array data can be loaded in stream load.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

This issue is a sub task of #7570 .

Array data can be loaded in stream load.

### Use case

```shell
mysql> create table array_type_table(k1 INT, k2 Array<int>) duplicate key (k1)
    -> distributed by hash(k1) buckets 1 properties('replication_num' = '1');
```

data.json
```json
[{"k1": 1, "v1": [1, 2, 3]}]
```

stream load:
```shell
curl -v --location-trusted -u root: -H "format: json" -T data.json http://<ip>:<port>/api/test/array_type_table/_stream_load
```

### Related issues

#7570 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by adonis0147**
# Proposed changes

Issue Number: close #8367 

## Problem Summary:

Please refer to #8367 .

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: Yes
3. Has document been added or modified: No
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/util/array_parser.hpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef DORIS_BE_SRC_UTIL_ARRAY_PARSER_HPP
+#define DORIS_BE_SRC_UTIL_ARRAY_PARSER_HPP
```
Comment (morningman) [inline be/src/util/array_parser.hpp:19]:
```suggestion
#pragma once
```

File: be/src/util/array_parser.hpp [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+            return Status::RuntimeError("Failed to parse the json to array.");
+        }
+        auto type_desc = convertToTypeDescriptor(context->get_return_type());
+        return parse<rapidjson::UTF8<>>(
+                array_val, context,
+                reinterpret_cast<const rapidjson::Document*>(&document)->GetArray(), type_desc);
+    }
+
+private:
+    static TypeDescriptor convertToTypeDescriptor(FunctionContext::TypeDesc function_type_desc) {
```
Comment (morningman) [inline be/src/util/array_parser.hpp:61]:
```suggestion
    static TypeDescriptor _convert_to_type_descriptor(FunctionContext::TypeDesc function_type_desc) {
```

Comment (adonis0147) [reply be/src/util/array_parser.hpp]: Done. PTAL @morningman .

File: be/src/util/array_parser.hpp [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+    static Status parse(CollectionVal& array_val, FunctionContext* context,
+                        const StringVal& str_val) {
+        rapidjson::Document document;
+        if (document.Parse(reinterpret_cast<char*>(str_val.ptr), str_val.len).HasParseError()) {
+            return Status::RuntimeError("Failed to parse the json to array.");
+        }
+        if (document.IsNull()) {
+            array_val = CollectionVal::null();
+            return Status::OK();
+        } else if (!document.IsArray()) {
```
Comment (HappenLee) [inline be/src/util/array_parser.hpp:50]: document.Parse(reinterpret_cast<char*>(str_val.ptr), str_val.len).HasParseError() || !document.IsArray()

File: be/src/util/array_parser.hpp [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            return Status::RuntimeError("Failed to parse the json to array.");
+        }
+        auto type_desc = _convertToTypeDescriptor(context->get_return_type());
+        return _parse<rapidjson::UTF8<>>(
+                array_val, context,
+                reinterpret_cast<const rapidjson::Document*>(&document)->GetArray(), type_desc);
+    }
+
+private:
+    static TypeDescriptor _convertToTypeDescriptor(FunctionContext::TypeDesc function_type_desc) {
```
Comment (HappenLee) [inline be/src/util/array_parser.hpp:60]: Do not use hump nomenclature

File: be/src/util/array_parser.hpp (hunk 3 above)
Comment (adonis0147) [reply be/src/util/array_parser.hpp]: Done.

File: be/src/util/array_parser.hpp (hunk 4 above)
Comment (adonis0147) [reply be/src/util/array_parser.hpp]: Done.

File: be/src/util/array_parser.hpp (hunk 1 above)
Comment (adonis0147) [reply be/src/util/array_parser.hpp]: Fixed.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**